### PR TITLE
chore: add stop subcommand

### LINKED ISSUE #741: Implmement sub-command like stop, restart for manager-api
- Why do you submit this issue?
- [ ] Question or discussion
- [ ] Bug
- [ ] Requirements
- [x] Feature or performance improvement
- [ ] Other

### Requirements or improvement.

Currently the restart of manage-api is manual, 。i.e. kill the process firstly, that's not so convenient, we can implement some sub command like `./manager-api stop` and `./manager-api restart` and etc (after https://github.com/apache/apisix-dashboard/issues/739 is solved).

### PR DESCRIPTION
**Pull request by tokers**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues

close #741 
___
### Bugfix
- Description

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

___
### Please add the corresponding test cases if necessary.

___
### Backport patches
- Why need to backport?

- Source branch

- Related commits and pull requests

- Target branch

### CODE REVIEW

**Comment by imjoey** [comment]
@tokers , thank you for this. Just FYI, the file `api/manager` is in this PR. 😄

File: api/internal/utils/pid.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+)
+
+// WritePID write pid to the given file path.
+func WritePID(filepath string) error {
```
Comment (nic-chen) [inline api/internal/utils/pid.go:28]: could we get the file path in the function since the function name is specified `WritePID` ?

**Comment by nic-chen** [comment]
> @tokers , thank you for this. Just FYI, the file `api/manager` is in this PR. 😄

yes, we should not submit the binary

**Comment by tokers** [comment]
> @tokers , thank you for this. Just FYI, the file `api/manager` is in this PR. 😄

Oh my gosh, i'll remove it, sorry~

File: api/internal/utils/pid.go (hunk 1 above)
Comment (tokers) [reply api/internal/utils/pid.go:28]: If we can sure the pid path in the future will not change, we can do it of course.

File: api/cmd/managerapi.go [hunk 2]
```diff
@@ -119,5 +132,28 @@ func NewManagerAPICommand() *cobra.Command {
...
+	cmd := &cobra.Command{
+		Use: "stop",
+		Run: func(cmd *cobra.Command, args []string) {
+			pid, err := utils.ReadPID(conf.PIDPath)
+			if err != nil {
+				if syscall.ENOENT.Error() == err.Error() {
+					fmt.Fprintf(os.Stderr, "failed to get manager-api pid: %s\n", err)
+				} else {
+					fmt.Fprintf(os.Stderr,  "pid path %s not found, is manager-api running?\n", conf.PIDPath)
+				}
```
Comment (starsz) [inline api/cmd/managerapi.go:150]: I think we should log the error info here.

`syscall.ENOENT.Error() == err.Error()` ? 
Maybe `syscall.ENOENT.Error() != err.Error()`

**Review by starsz — CHANGES_REQUESTED** [review]

**Comment by idbeta** [comment]
You should add instructions in the document.

**Review by membphis — COMMENTED** [review]
need to update the doc, help the user know this command

File: api/cmd/managerapi.go (hunk 2 above)
Comment (tokers) [reply api/cmd/managerapi.go:150]: Yes.

Comment (tokers) [reply api/cmd/managerapi.go:150]: Fixed.

**Comment by membphis** [comment]
ping @tokers

**Comment by tokers** [comment]
Ok, will add it as soon as Possible.

**Comment by tokers** [comment]
@membphis @idbeta Added.

**Comment by membphis** [comment]
> @membphis @idbeta Added.

LGTM

**Review by nic-chen — APPROVED** [review]

**Comment by juzhiyuan** [comment]
ping @starsz @imjoey

**Comment by idbeta** [comment]
> @membphis @idbeta Added.
LGTM

**Comment by juzhiyuan** [comment]
ping @tokers

**Review by imjoey — APPROVED** [review]
/lgtm

Thanks @tokers .

**Review by starsz — APPROVED** [review]

**Pull request merged**